crates/fluvio-cli/src/topic/create.rs

```diff
@@ -121,10 +121,15 @@ impl CreateTopicOpt {
             })
         };
 
-        let is_valid = hostname_validator::is_valid(&self.topic);
+        let is_valid = self
+            .topic
+            .chars()
+            .all(|ch| ch.is_ascii_lowercase() || ch.is_ascii_digit() || ch == '-')
+            && !self.topic.ends_with('-')
+            && !self.topic.starts_with('-');
         if !is_valid {
             return Err(CliError::InvalidArg(
-                "Topic name must only contain alphanumeric characters, '-', or '.'".to_string(),
+                "Topic name must only contain lowercase alphanumeric characters or '-', start with an alphanumeric character and end with an alphanumeric character.".to_string(),
```

Comment:
let's move this message to Error

Reply:
this should move to client error

Reply:
Not sure what you mean, in the client side this is wrapped behind the ApiError type with the status code and message https://github.com/infinyon/fluvio/blob/122fb9fd01a7dc6f89bb3c15285b9ca86c0d5add/crates/fluvio-sc-schema/src/response.rs#L46

Reply:
Error Message should be coming from ErrorCode (canonical).  ErrrorCode->ApiError->AdminError (Fluvio crate).

In the Fluvio Error, we have:
```
 #[error("Fluvio controlplane error: {0}")]
    AdminApi(#[from] ApiError),
```
We should just forward error message

Reply:
Got it, in this particular case, this error is from CLI parsing arguments, not sure if that applies to this, but I will change this on the SC handling side to include only the error code